spec/queue.js

```diff
@@ -19,4 +19,87 @@ describe('Queue', () => {
         .to.alter(() => myQueue.length(), { from: 0, to: 1 })
     })
   })
+
+  context('dequeue()', () => {
+    it('returns and removes the front element in the queue or null if the queue is empty.', () => {
+      const myQueue = new Queue()
+
+      myQueue.enqueue('foo')
+      myQueue.enqueue('bar')
+      expect( myQueue.dequeue() ).to.equal('foo')
+    })
+  })
+
+  context('dequeue()', () => {
+    it('returns null on an empty queue.', () => {
+      const myQueue = new Queue()
+
+      expect( myQueue.dequeue() ).to.equal(null)
+    })
+  })
+
+  context('front()', () => {
```

Comment:
Consider refactoring to use `context` to describe the variants (when empty, when items), as above.